### DISCUSSION TITLE
CS0165 Use of unassigned local variable after [DoesNotReturn] method

### DISCUSSION THREAD

**Discussion by D0ctorWh0** [dotnet/csharplang#6604] [General]
If I have method, that never return

```
[DoesNotReturn]
public void Exit()
{
    Environment.Exit(1);
}
```

than if I write this code
```
int? i = 0;

if(i is not int val)
{
    Exit();
}

int t = val;
```

the compiler shows error CS0165 "Use of unassigned local variable" on line `int t = val`;. But as "Exit" method never returns it should not show error. The workaround for it is to add `return;` after calling "Exit" method, but that not feel right...

**Comment by 333fred** [comment]
See https://github.com/dotnet/csharplang/issues/538. The DoesNotReturnAttribute cannot give the guarantees needed to make this work.
